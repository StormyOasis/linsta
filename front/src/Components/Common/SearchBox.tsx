import React from "react";
import styled from "styled-components";
import { Flex} from "../Common/CombinedStyling";
import { CircleXSVG, SearchBoxSVG } from "./Icon";


const SearchContainer = styled(Flex)`
    align-items: center;
    background-color: ${props => props.theme["colors"].buttonDefaultSecondaryColor};
    border: 1px solid ${props => props.theme["colors"].borderDefaultColor};
    border-radius: 8px;
    padding: 0.5rem 0.75rem;
    width: 100%;
`;

const Input = styled.input`
    border: none;
    outline: none;
    font-size: 1rem;
    width: 100%;
    background-color: ${props => props.theme["colors"].buttonDefaultSecondaryColor};
`;

const IconWrapper = styled(Flex)`
    align-items: center;
    margin-right: 0.5rem;
`;

const ClearButton = styled.button`
    background: none;
    border: none;
    cursor: pointer;
    padding: 0;
    margin-left: 0.5rem;
    font-size: 1rem;
    color: ${props => props.theme["colors"].borderDarkColor};

    &:hover {
        color: ${props => props.theme["colors"].borderDefaultColor};
    }
`;

type SearchBoxProps = {
    placeholder?: string;
    value: string;
    onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
    onClear: () => void;
};

const SearchBox: React.FC<SearchBoxProps> = (props: SearchBoxProps) => {
    return (
        <SearchContainer>
            <IconWrapper>
                <SearchBoxSVG width="16px" height="16px" />
            </IconWrapper>            
            <Input
                type="text"
                placeholder={props.placeholder || "Search..."}
                value={props.value}
                onChange={props.onChange}
            />
            {props.value && (
                <ClearButton onClick={props.onClear} aria-label="Clear search">
                    <CircleXSVG width="16px" height="16px" />
                </ClearButton>
            )}            
        </SearchContainer>
    );
}

export default SearchBox;