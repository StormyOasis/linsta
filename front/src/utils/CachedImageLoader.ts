import { clear, del, get, set } from 'idb-keyval';
import { blobToBase64 } from "./utils";

export const loadImageCached = async (id: string, url: string): Promise<string> => {
    // First try to get the image from the cache
    const value = await get(id);

    if(value === undefined) {
        // Image not in the cache so have to fetch it
        try {
            const result = await fetch(url);
            if(result.status === 200) {
                const blob = await result.blob();
                const b64:string = await blobToBase64(blob) as string;

                // add the base64 data into the cache and return it
                await set(id, b64);

                return b64;
            }
        } catch(err) {
            console.log(err);
            return url; //default to what was passed in
        }

    } else {
        // Image is in the cache already as a base64 string, use that as is
        return value;
    }

    return url; //default to what was passed in
}

export const setImage = async (id: string, data: string) => {     
    await set(id, data);
}

export const deleteFromCache = async (id: string) => {
    // First try to get the image from the cache
    const value = await get(id);

    if(value === undefined) {
        // key not present in cache so do nothing
        return;
    }

    // delete from the cache
    await del(id);
}

export const clearCache = async () => {
    await clear();
}