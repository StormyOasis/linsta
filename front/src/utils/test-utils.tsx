import React, { ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { RootState } from "../Components/Redux/redux";
import { authInitialState } from '../Components/Redux/slices/auth.slice';
import { modalInitialState } from '../Components/Redux/slices/modals.slice';
import { profileInitialState } from '../Components/Redux/slices/profile.slice';
import { miscInitialState } from '../Components/Redux/slices/misc.slice';
import { configureStore } from '@reduxjs/toolkit';

import Theme from '../Components/Themes/Theme';

export const mockInitialState = (overrides?: Partial<RootState>): RootState => ({
    auth: { ...authInitialState, ...(overrides?.auth ?? {}) },
    modal: { ...modalInitialState, ...(overrides?.modal ?? {}) },
    profile: { ...profileInitialState, ...(overrides?.profile ?? {}) },
    misc: { ...miscInitialState, ...(overrides?.misc ?? {}) },
});

type ExtendedRenderOptions = {
    store?: ReturnType<typeof configureStore>;
    preloadedState?: any;
    route?: string; // ‚Üê allow overriding the initial route
} & Omit<RenderOptions, "queries">;

const dummyAuthReducer = (state = { user: null, isLoading: false, error: null }, _action: any) => state;
const dummyModalReducer = (state = { isOpen: false }, _action: any) => state;
const dummyProfileReducer = (state = { name: '', email: '' }, _action: any) => state;
const dummyMiscReducer = (state = { someData: null }, _action: any) => state;

const renderWithStore = (
    ui: ReactElement,
    {
        preloadedState,
        store = configureStore({
            reducer: {
                auth: dummyAuthReducer,
                modal: dummyModalReducer,
                profile: dummyProfileReducer,
                misc: dummyMiscReducer,
            },
            preloadedState,
        }),
        route = "/",
        ...renderOptions
    }: ExtendedRenderOptions = {}
) => {
    const Wrapper: React.FC<{ children: React.ReactNode }> = ({ children }) => {
        return (
            <Provider store={store}>
                <MemoryRouter initialEntries={[route]}>
                    <Theme>{children}</Theme>
                </MemoryRouter>
            </Provider>
        );
    };

    return {
        ...render(ui, { wrapper: Wrapper, ...renderOptions }),
        store,
    };
};

export * from "@testing-library/react";
export { renderWithStore };
export type { ExtendedRenderOptions };