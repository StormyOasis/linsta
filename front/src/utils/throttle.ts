import { useRef } from 'react';

const useThrottle = (callback: Function, delay: number) => {
    const lastCall = useRef(0);

    return (...args: any[]) => {
        const now = new Date().getTime();
        if (now - lastCall.current >= delay) {
            lastCall.current = now;
            callback(...args);
        }
    };
};

export default useThrottle;