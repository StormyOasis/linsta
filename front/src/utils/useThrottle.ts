import { useRef, useCallback } from 'react';

/**
 * React hook for throttling a callback function.
 * The callback will only be invoked at most once every `delay` milliseconds.
 *
 * @param callback - The function to throttle.
 * @param delay - The minimum delay (ms) between calls.
 * @returns A throttled version of the callback.
 */
const useThrottle = <T extends (...args: any[]) => void>(callback: T, delay: number) => {
    const lastCall = useRef<number>(0);

    return useCallback(
        (...args: Parameters<T>) => {
            const now = performance.now();
            if (now - lastCall.current >= delay) {
                lastCall.current = now;
                callback(...args);
            }
        },
        [callback, delay]
    );
};

export default useThrottle;