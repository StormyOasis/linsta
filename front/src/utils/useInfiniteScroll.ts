import React, { useEffect, useRef } from 'react';
import useThrottle from './useThrottle';

/**
 * Type for the async load more function.
 */
type AsyncLoadMoreFunction = () => Promise<void>;

/**
 * Custom hook for infinite scrolling.
 * Attaches a throttled scroll event listener to a container and triggers `loadMore`
 * when the user scrolls near the bottom.
 *
 * @param loadMore - Async function to load more items.
 * @param isLoading - Whether a load is currently in progress.
 * @param containerRef - Ref to the scrollable container element.
 */
const useInfiniteScroll = (loadMore: AsyncLoadMoreFunction, 
    isLoading: boolean, 
    childRef: React.MutableRefObject<HTMLDivElement | null>) => {
        
    const hasScrolled = useRef<boolean>(false);  // Track if the user has scrolled

    // Throttled scroll handler
    const throttledHandleScroll = useThrottle(async () => {
        if (typeof window !== 'undefined' && childRef.current) {
            const element = childRef.current as HTMLElement;
            const currentScroll = window.innerHeight + element.scrollTop;

            if (currentScroll + 256 >= element.scrollHeight && !isLoading && !hasScrolled.current) {
                hasScrolled.current = true;  // Mark as scrolled at least once
                await loadMore();
            }
        }
    }, 200);

    useEffect(() => {
        const childElement = childRef.current;
        if (childElement) {
            childElement.addEventListener('scroll', throttledHandleScroll);
        }

        // Clean up event listener on unmount
        return () => {
            if (childElement) {
                childElement.removeEventListener('scroll', throttledHandleScroll);
            }
        };
    }, [throttledHandleScroll, childRef]);

    // Reset trigger flag when loading finishes
    useEffect(() => {
        if (!isLoading) {
            hasScrolled.current = false;
        }
    }, [isLoading]);
}

export default useInfiniteScroll;