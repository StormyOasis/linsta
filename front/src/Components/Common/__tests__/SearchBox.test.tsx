import React from "react";
import { screen, fireEvent } from "@testing-library/react";
import SearchBox from "../SearchBox";
import '@testing-library/jest-dom';
import { renderWithStore } from '../../../utils/test-utils';

// Mock icons
jest.mock("../Icon", () => ({
    CircleXSVG: (props: any) => <svg data-testid="circle-x-svg" {...props} />,
    SearchBoxSVG: (props: any) => <svg data-testid="search-box-svg" {...props} />,
}));

describe("SearchBox", () => {
    const baseProps = {
        placeholder: "Search users...",
        value: "",
        onChange: jest.fn(),
        onClear: jest.fn(),
    };

    it("renders input and search icon", () => {
        const r = renderWithStore(<SearchBox {...baseProps} />);
        expect(screen.getByPlaceholderText("Search users...")).toBeInTheDocument();
        expect(screen.getByTestId("search-box-svg")).toBeInTheDocument();
        expect(r).toMatchSnapshot();
    });

    it("shows clear button when value is not empty", () => {
        const r = renderWithStore(<SearchBox {...baseProps} value="alice" />);
        expect(screen.getByTestId("circle-x-svg")).toBeInTheDocument();
        expect(screen.getByLabelText("Clear search")).toBeInTheDocument();
        expect(r).toMatchSnapshot();
    });

    it("does not show clear button when value is empty", () => {
        const r = renderWithStore(<SearchBox {...baseProps} value="" />);
        expect(screen.queryByTestId("circle-x-svg")).not.toBeInTheDocument();
        expect(screen.queryByLabelText("Clear search")).not.toBeInTheDocument();
        expect(r).toMatchSnapshot();
    });

    it("calls onChange when input changes", () => {
        const r = renderWithStore(<SearchBox {...baseProps} />);
        const input = screen.getByPlaceholderText("Search users...");
        fireEvent.change(input, { target: { value: "bob" } });
        expect(baseProps.onChange).toHaveBeenCalled();
        expect(r).toMatchSnapshot();
    });

    it("calls onClear when clear button is clicked", () => {
        const r = renderWithStore(<SearchBox {...baseProps} value="alice" />);
        const clearBtn = screen.getByLabelText("Clear search");
        fireEvent.click(clearBtn);
        expect(baseProps.onClear).toHaveBeenCalled();
        expect(r).toMatchSnapshot();
    });
});