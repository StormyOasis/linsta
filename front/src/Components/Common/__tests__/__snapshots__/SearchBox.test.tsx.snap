// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox calls onChange when input changes 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 SearchBox__SearchContainer-sc-7yk9ly-0 escqfF ckFvxY kdkliK"
      >
        <div
          class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 SearchBox__IconWrapper-sc-7yk9ly-2 escqfF ckFvxY dpzfPM"
        >
          <svg
            data-testid="search-box-svg"
            fill="currentColor"
            height="16px"
            stroke="none"
            stroke-width="1"
            width="16px"
          />
        </div>
        <input
          class="SearchBox__Input-sc-7yk9ly-1 dhtbre"
          placeholder="Search users..."
          type="text"
          value=""
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 SearchBox__SearchContainer-sc-7yk9ly-0 escqfF ckFvxY kdkliK"
    >
      <div
        class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 SearchBox__IconWrapper-sc-7yk9ly-2 escqfF ckFvxY dpzfPM"
      >
        <svg
          data-testid="search-box-svg"
          fill="currentColor"
          height="16px"
          stroke="none"
          stroke-width="1"
          width="16px"
        />
      </div>
      <input
        class="SearchBox__Input-sc-7yk9ly-1 dhtbre"
        placeholder="Search users..."
        type="text"
        value=""
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`SearchBox calls onClear when clear button is clicked 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 SearchBox__SearchContainer-sc-7yk9ly-0 escqfF ckFvxY kdkliK"
      >
        <div
          class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 SearchBox__IconWrapper-sc-7yk9ly-2 escqfF ckFvxY dpzfPM"
        >
          <svg
            data-testid="search-box-svg"
            fill="currentColor"
            height="16px"
            stroke="none"
            stroke-width="1"
            width="16px"
          />
        </div>
        <input
          class="SearchBox__Input-sc-7yk9ly-1 dhtbre"
          placeholder="Search users..."
          type="text"
          value="alice"
        />
        <button
          aria-label="Clear search"
          class="SearchBox__ClearButton-sc-7yk9ly-3 cRyWTB"
        >
          <svg
            data-testid="circle-x-svg"
            fill="currentColor"
            height="16px"
            stroke="none"
            stroke-width="1"
            width="16px"
          />
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 SearchBox__SearchContainer-sc-7yk9ly-0 escqfF ckFvxY kdkliK"
    >
      <div
        class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 SearchBox__IconWrapper-sc-7yk9ly-2 escqfF ckFvxY dpzfPM"
      >
        <svg
          data-testid="search-box-svg"
          fill="currentColor"
          height="16px"
          stroke="none"
          stroke-width="1"
          width="16px"
        />
      </div>
      <input
        class="SearchBox__Input-sc-7yk9ly-1 dhtbre"
        placeholder="Search users..."
        type="text"
        value="alice"
      />
      <button
        aria-label="Clear search"
        class="SearchBox__ClearButton-sc-7yk9ly-3 cRyWTB"
      >
        <svg
          data-testid="circle-x-svg"
          fill="currentColor"
          height="16px"
          stroke="none"
          stroke-width="1"
          width="16px"
        />
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`SearchBox does not show clear button when value is empty 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 SearchBox__SearchContainer-sc-7yk9ly-0 escqfF ckFvxY kdkliK"
      >
        <div
          class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 SearchBox__IconWrapper-sc-7yk9ly-2 escqfF ckFvxY dpzfPM"
        >
          <svg
            data-testid="search-box-svg"
            fill="currentColor"
            height="16px"
            stroke="none"
            stroke-width="1"
            width="16px"
          />
        </div>
        <input
          class="SearchBox__Input-sc-7yk9ly-1 dhtbre"
          placeholder="Search users..."
          type="text"
          value=""
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 SearchBox__SearchContainer-sc-7yk9ly-0 escqfF ckFvxY kdkliK"
    >
      <div
        class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 SearchBox__IconWrapper-sc-7yk9ly-2 escqfF ckFvxY dpzfPM"
      >
        <svg
          data-testid="search-box-svg"
          fill="currentColor"
          height="16px"
          stroke="none"
          stroke-width="1"
          width="16px"
        />
      </div>
      <input
        class="SearchBox__Input-sc-7yk9ly-1 dhtbre"
        placeholder="Search users..."
        type="text"
        value=""
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`SearchBox renders input and search icon 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 SearchBox__SearchContainer-sc-7yk9ly-0 escqfF ckFvxY kdkliK"
      >
        <div
          class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 SearchBox__IconWrapper-sc-7yk9ly-2 escqfF ckFvxY dpzfPM"
        >
          <svg
            data-testid="search-box-svg"
            fill="currentColor"
            height="16px"
            stroke="none"
            stroke-width="1"
            width="16px"
          />
        </div>
        <input
          class="SearchBox__Input-sc-7yk9ly-1 dhtbre"
          placeholder="Search users..."
          type="text"
          value=""
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 SearchBox__SearchContainer-sc-7yk9ly-0 escqfF ckFvxY kdkliK"
    >
      <div
        class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 SearchBox__IconWrapper-sc-7yk9ly-2 escqfF ckFvxY dpzfPM"
      >
        <svg
          data-testid="search-box-svg"
          fill="currentColor"
          height="16px"
          stroke="none"
          stroke-width="1"
          width="16px"
        />
      </div>
      <input
        class="SearchBox__Input-sc-7yk9ly-1 dhtbre"
        placeholder="Search users..."
        type="text"
        value=""
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`SearchBox shows clear button when value is not empty 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 SearchBox__SearchContainer-sc-7yk9ly-0 escqfF ckFvxY kdkliK"
      >
        <div
          class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 SearchBox__IconWrapper-sc-7yk9ly-2 escqfF ckFvxY dpzfPM"
        >
          <svg
            data-testid="search-box-svg"
            fill="currentColor"
            height="16px"
            stroke="none"
            stroke-width="1"
            width="16px"
          />
        </div>
        <input
          class="SearchBox__Input-sc-7yk9ly-1 dhtbre"
          placeholder="Search users..."
          type="text"
          value="alice"
        />
        <button
          aria-label="Clear search"
          class="SearchBox__ClearButton-sc-7yk9ly-3 cRyWTB"
        >
          <svg
            data-testid="circle-x-svg"
            fill="currentColor"
            height="16px"
            stroke="none"
            stroke-width="1"
            width="16px"
          />
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 SearchBox__SearchContainer-sc-7yk9ly-0 escqfF ckFvxY kdkliK"
    >
      <div
        class="CombinedStyling__Div-sc-c3jdxk-0 CombinedStyling__Flex-sc-c3jdxk-2 SearchBox__IconWrapper-sc-7yk9ly-2 escqfF ckFvxY dpzfPM"
      >
        <svg
          data-testid="search-box-svg"
          fill="currentColor"
          height="16px"
          stroke="none"
          stroke-width="1"
          width="16px"
        />
      </div>
      <input
        class="SearchBox__Input-sc-7yk9ly-1 dhtbre"
        placeholder="Search users..."
        type="text"
        value="alice"
      />
      <button
        aria-label="Clear search"
        class="SearchBox__ClearButton-sc-7yk9ly-3 cRyWTB"
      >
        <svg
          data-testid="circle-x-svg"
          fill="currentColor"
          height="16px"
          stroke="none"
          stroke-width="1"
          width="16px"
        />
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
